cassandra_migrate/migrator.py

```diff
@@ -384,42 +386,68 @@ def _create_version(self, version, migration):
 
         return version_id
 
-    def _apply_migration(self, version, migration, skip=False):
+    def _apply_cql_migration(self, version, migration):
         """
-        Persist and apply a migration
+        Persist and apply a cql migration
 
         First create an in-progress version entry, apply the script, then
         finalize the entry as succeeded, failed or skipped.
+        """
+
+        self.logger.info('Applying cql migration')
+
+        statements = CqlSplitter.split(migration.content)
+
+        try:
+            if statements:
+                self.logger.info('Executing migration -'
+                                 '{} CQL statement'.format(len(statements)))
+
+            for statement in statements:
+                self.session.execute(statement)
+        except Exception:
+            self.logger.exception('Failed to execute migration')
+            raise FailedMigration(version, migration.name)
+
+    def _apply_python_migration(self, version, migration):
+        """
+        Persist and apply a python migration
+
+        First create an in-progress version entry, apply the script, then
+        finalize the entry as succeeded, failed or skipped.
+        """
+        self.logger.info('Applying python script')
+
+        try:
+            check_call([sys.executable, migration.path])
+        except Exception:
+            self.logger.exception('Failed to execute script')
+            raise FailedMigration(version, migration.name)
+
+    def _apply_migration(self, version, migration, skip=False):
+        """
+        Persist and apply a migration
 
         When `skip` is True, do everything but actually run the script, for
         example, when baselining instead of migrating.
         """
 
         self.logger.info('Advancing to version {}'.format(version))
 
-        if skip:
-            statements = []
-            self.logger.info('Migration is marked for skipping, '
-                             'not actually running script')
-        else:
-            statements = CqlSplitter.split(migration.content)
-
         version_uuid = self._create_version(version, migration)
         new_state = Migration.State.FAILED
 
         result = None
-        try:
-            if statements:
-                self.logger.info('Executing migration - '
-                                 '{} CQL statements'.format(len(statements)))
-
-            # Set default keyspace so migrations don't need to refer to it
-            # manually
-            # Fixes https://github.com/Cobliteam/cassandra-migrate/issues/5
-            self.session.execute('USE {};'.format(self.config.keyspace))
 
-            for statement in statements:
-                self.session.execute(statement)
+        try:
+            if skip:
+                self.logger.info('Migration is marked for skipping, '
+                                 'not actually running script')
+            else:
+                if migration.filetype:
```

Comment:
The attribute name seems somewhat misleading here. I would expect filetype to always be filled with either `cql` or `py`, or for it to be named `is_python` or something of the sort.

Reply:
Daniel, 

This is one of those 'rookie' mistakes leftovers. Initially I was verifying the content of that object... then I realize that I can treat it like a boolean (if it comes empty because regex didn't find a match). So yes, in that context I agree it looks like a boolean from its use in the `if` statement. 

I'll fix that.